ui/pages/login.tsx

```diff
@@ -171,12 +184,38 @@ const Login: NextPage = () => {
   };
 
   let isWebauthn = false;
+  let isPasswordOnly = false;
   const supportsWebauthn = flow?.ui.nodes.some(
     (node) => node.group === "webauthn",
   );
   const renderFlow = isAuthCode ? filterFlow(replaceAuthLabel(flow)) : flow;
 
   if (renderFlow?.ui) {
+    isPasswordOnly =
+      renderFlow.ui.messages?.length === 1 &&
+      renderFlow.ui.messages[0].id === 1010003;
+    if (isPasswordOnly) {
+      renderFlow.ui.nodes = renderFlow.ui.nodes.map((node) => {
+        if (
+          node.group !== "default" ||
+          (node.attributes as UiNodeInputAttributes).name !== "identifier"
+        ) {
+          return node;
+        }
+
```

Comment:
can we wrap this 2 pieces of logic in two separate function with an explicative name?